heron/spi/src/java/com/twitter/heron/spi/utils/TMasterUtils.java

```diff
@@ -114,49 +114,51 @@ private static boolean sendGetRequest(URL endpoint, String command,
    */
   private static TopologyAPI.TopologyState getRuntimeTopologyState(
       String topologyName,
-      SchedulerStateManagerAdaptor statemgr) {
+      SchedulerStateManagerAdaptor statemgr) throws TMasterException {
     PhysicalPlans.PhysicalPlan plan = statemgr.getPhysicalPlan(topologyName);
 
     if (plan == null) {
-      LOG.log(Level.SEVERE, "Failed to get physical plan for topology {0}", topologyName);
-      return null;
+      String errMsg = String.format("Failed to get physical plan for topology '%s'", topologyName);
+      throw new TMasterException(errMsg);
     }
 
     return plan.getTopology().getState();
   }
 
-  public static boolean transitionTopologyState(String topologyName,
+  public static void transitionTopologyState(String topologyName,
                                                 TMasterCommand topologyStateControlCommand,
                                                 SchedulerStateManagerAdaptor statemgr,
                                                 TopologyAPI.TopologyState startState,
                                                 TopologyAPI.TopologyState expectedState,
-                                                NetworkUtils.TunnelConfig tunnelConfig) {
+                                                NetworkUtils.TunnelConfig tunnelConfig)
+  throws TMasterException {
     TopologyAPI.TopologyState state = TMasterUtils.getRuntimeTopologyState(topologyName, statemgr);
     if (state == null) {
-      LOG.severe("Topology still not initialized.");
-      return false;
+      throw new TMasterException(String.format(
+          "Topology '%s' is not initialized yet", topologyName));
     }
 
     if (state == expectedState) {
-      LOG.log(Level.SEVERE, "Topology {0} command received topology {1} but already in {2} state",
-          new Object[] {topologyStateControlCommand, topologyName, state});
-      return true;
+      String errMsg = String.format(
+          "Topology {0} command received topology {1} but already in {2} state",
+          topologyStateControlCommand, topologyName, state);
+      throw new TMasterException(errMsg);
```

Comment:
I favor using Preconditions.checkState for error states that should never really occur except due to a bug (i.e. dividing by 0 for example). For situations like this one though, this can actually happen if two people are trying to run a command at the same time. In this cass I think it's good to have a better typed exception.